api/cluster.go

```diff
@@ -153,3 +186,69 @@ func (s *Server) indexDelete(ctx *middleware.Context, req models.IndexDelete) {
 	}
 	response.Write(ctx, response.NewMsgp(200, &resp))
 }
+
+// peerQuery takes a request and the path to request it on, then fans it out
+// across the cluster, except to the local peer.
+// ctx:          request context
+// data:         request to be submitted
+// name:         name to be used in logging & tracing
+// path:         path to request on
+// respTemplate: the response object into which the responses shall
+//               be deserialized. each sub-request will get a copy of this
+//               template and the response will be unarshalled into that copy.
+//               (generic without generics)
```

Comment:
this seems incorrect.
doing `resp := respTemplate` is not al copy. see https://play.golang.org/p/vILbXj3BfU for a demo.
so seems to me like this will corrupt the responses of different nodes, effectively what would happen I think is that all the responses are the response of the last node to respond.

Reply:
(the demo is based on the code for Server.clusterTagDetails )

Reply:
yeah true. i guess that way of making it generic, by passing in a template to copy, is kind of ugly anyway. what would be better is to first get the list of peers in the caller, generate the slice of responses based on the number of peers, and then pass both slices into `peerQuery()`

Reply:
that above suggestion doesn't work because `[]<specific type>` will not be accepted if the signature expects `[]interface`: https://github.com/golang/go/wiki/InterfaceSlice
but what do you think about that: https://github.com/grafana/metrictank/pull/750/commits/18b2257206765b49ec25c0a30a93e91f2332e4d2

Reply:
> because []<specific type> will not be accepted if the signature expects []interface

yeah bummer.  i've also run into this before.

> but what do you think about that: 18b2257

it's unfortunate that we have to move half of peerQueryi's processing logic back into the callers. that starts to defeat the point of the reusable function. i wonder if the actual type can be passed in and then peerQuery could instantiate however many of the type it needs via https://golang.org/pkg/reflect/#New

Reply:
i think that would be nice. but without knowing it, just based on what i understand the `reflect` lib does, i'd guess that's incurring overhead